ecs-init/apparmor/apparmor_utils.go

```diff
@@ -0,0 +1,68 @@
+/*
+   Copyright The docker Authors.
+   Copyright The Moby Authors.
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package apparmor
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	exec "golang.org/x/sys/execabs"
+)
+
+// NOTE: This code is copied from <github.com/docker/docker/profiles/apparmor>.
+//       If you plan to make any changes, please make sure they are also sent
+//       upstream.
```

Comment:
Is there any significant reason we don't just consume the upstream code directly?

If there is such a reason and we need to keep this file, do we want to add some unit tests for this file? The test coverage for it is currently at 0%.

Reply:
The only reason we dont consume from upstream is that the functions are all private, so we can't import it

Reply:
In github.com/docker/docker/profiles/apparmor. I could only find the corresponding `isLoaded` function, which seems to be public:
https://github.com/moby/moby/blob/master/profiles/apparmor/apparmor.go#L102

Could you point to where `load` and `aaParser` are copied from? Were they copied from `github.com/docker/docker/pkg/aaparser` instead of github.com/docker/docker/profiles/apparmor? In the former I did find a function `LoadProfile` that looks like it may correspond to `load`, but that function is also public: https://github.com/moby/moby/blob/master/pkg/aaparser/aaparser.go#L30

Reply:
Ah gotcha. If that is the case, could we consume the upstream code that containerd copied from in the first place and made private? That way we remain in sync with any upstream updates/bug fixes that are made, and we can trust that any functions we use were tested upstream before being released. As a result, we would not have to maintain a mirror file nor write unit tests for the same.

Reply:
I agree, updated to consume from:
`github.com/docker/docker/pkg/aaparser`
`github.com/docker/docker/profiles/apparmor`